devito/passes/iet/mpi.py

```diff
@@ -368,3 +427,73 @@ def mpiize(graph, **kwargs):
         make_halo_exchanges(graph, mpimode=mpimode, **kwargs)
 
     make_reductions(graph, mpimode=mpimode, **kwargs)
+
+
+# Utility functions to avoid code duplication
+
+def denest_halospots(iet):
+    """
+    Denest nested HaloSpots.
+    # TOFIX: This also merges HaloSpots that have different loc_indices
```

Comment:
This is not a problem anymore, but I could not find some solution/custom idea that can trigger that even after the new pass